tests/moe/test_trtllm_gen_routed_fused_moe.py

```
@@ -0,0 +1,248 @@
+)
+@pytest.mark.parametrize("quant_mode", ["MxFP4xMxFP8", "MxFP4xBf16"])
+def test_trtllm_gen_routed_fused_moe(
+    num_tokens: int,
+    hidden_size: int,
+    intermediate_size: int,
+    top_k: int,
+    num_experts: int,
+    routing_method_type: RoutingMethodType,
+    quant_mode: Literal["NvFP4xNvFP4", "MxFP4xMxFP8", "MxFP4xBf16"],
+):
+    if num_tokens == 1 or num_tokens == 8 and quant_mode == "NvFP4xNvFP4":
```

Ruff (0.14.3)

Parenthesize `a and b` expressions when chaining `and` and `or` together, to make the precedence clear

Parenthesize the `and` subexpression

(RUF021)